modules/base/rendering/screenspaceinsetblackout.cpp

```diff
@@ -0,0 +1,951 @@
+        "CopyToClipboard",
+        "Copy to clipboard",
+        "Copies the current configuration to the clipboard.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo CalibrationPatternInfo = {
+        "EnableCalibrationPattern",
+        "Enable Calibration Pattern",
+        "Enables the calibration pattern.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo CalibrationColorInfo = {
+        "EnableCalibrationColor",
+        "Enable Calibration Color",
+        "Set Blackout Shape to a bright color for easier calibration.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo AddControlPointInfo = {
+        "AddControlPoint",
+        "Add Control Point",
+        "Adds a new control point to the spline.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo RemoveControlPointInfo = {
+        "RemoveControlPoint",
+        "Remove Control Point",
+        "Removes the selected control point from the spline.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo RemoveSelectorInfo = {
+        "RemoveSelector",
+        "Select Point To Remove",
+        "Removes the selected control point.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo AddSelectorInfo = {
+        "AddSelector",
+        "Select Where To Add",
+        "Select where to add a new point.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo NewPointPositionInfo = {
+        "NewPointPosition",
+        "Point Position",
+        "X and Y coordinates for where to add the new control point.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo CalibrationTextureInfo = {
+        "CalibrationTexture",
+        "Calibration Texture",
+        "Texture used as calibration pattern.",
+        openspace::properties::Property::Visibility::Hidden
+    };
+
+    struct [[codegen::Dictionary(ScreenSpaceInsetBlackout)]] Parameters {
+        std::optional<std::string> identifier;
+        struct BlackoutShape {
+            std::vector<glm::vec2> corners;
+            std::optional<std::vector<glm::vec2>> top;
+            std::optional<std::vector<glm::vec2>> bottom;
+            std::optional<std::vector<glm::vec2>> left;
+            std::optional<std::vector<glm::vec2>> right;
+            std::optional<std::string> calibrationTexturePath;
+        };
+        BlackoutShape blackoutshape;
+    };
```

Comment:
Please add documentation to this struct. This is the documentation thaw will show up on the docs.openspaceproject.com page. 

In the following format: 

```suggestion
    // Add a multiline explanation of the renderable and what it can be used for. 
    // See for example navigationstate.cpp
    struct [[codegen::Dictionary(ScreenSpaceInsetBlackout)]] Parameters {
        // Add some docs here
        std::optional<std::string> identifier;
        
        struct BlackoutShape {
            // Add some docs here
            std::vector<glm::vec2> corners;
            
             // Add some docs here
            std::optional<std::vector<glm::vec2>> top;
                        
             // Add some docs here
            std::optional<std::vector<glm::vec2>> bottom;
                        
             // Add some docs here
            std::optional<std::vector<glm::vec2>> left;
                        
             // Add some docs here
            std::optional<std::vector<glm::vec2>> right;
                        
             // Add some docs here
            std::optional<std::string> calibrationTexturePath;
        };
                    
        // Add some docs here
        BlackoutShape blackoutshape;
    };
```

Reply:
Also note that `blackoutshape` should be `blackoutShape`   (But then you also need to change the asset you might have that creates this)